packages/headless/src/database/mongodb/order/index.ts

```diff
@@ -28,10 +29,19 @@ export class OrderDatabase implements IOrderDatabase {
       return newProductList;
   }
 
-  async getOrderListByUserId(userId: string): Promise<OrderEntity[]> {
+  async getOrderListByUserId(userId: string, sortObj: OrderSortQuery): Promise<OrderEntity[]> {
+    const { sortField, sortType} = sortObj;
+
     const orderList = await OrderModel.find({ userId });
-    if (orderList.length > 0) {
-      return orderList;
+    if (orderList.length > 0){
+      if(sortType === 'ascending'){
+        return orderList.sort((a,b) => a[sortField] - b[sortField]);
+      }else if(sortType === 'descending'){
+        return orderList.sort((a,b) => b[sortField] - a[sortField]);
+      }else{
+        const sortField: string = "orderedDate";
+        return orderList.sort((a,b) => b[sortField] - a[sortField]);
+      }
```

Comment:
this is not good coding. sorting data by javascript is not good idea. only use sorting using js where necessary and there's no other way.
but here mongodb has both sort and pagination support. you just need to know about that.